cmake/common.cmake

```diff
@@ -1,660 +1,679 @@
+
+        # We need to call sphinx-build with -E to make sure all files are regenerated.
+        COMMAND ${CMAKE_COMMAND} -E env SPHINX_BUILD_CONFIGURATION_FILE=${_reports_config_json} AUTOCONF_JSON_FILE=${AUTOCONF_JSON} VARIANT=${VARIANT} -- sphinx-build -E -b html ${PROJECT_SOURCE_DIR} ${_reports_html_output_dir}
+        BYPRODUCTS ${_reports_html_output_dir}/index.html
+        DEPENDS _components_variant_coverage_html_target ${_components_variant_doxysphinx_targets}
+    )
+endmacro()
+
+macro(_spl_set_coverage_create_overall_report_is_necessary)
+    set(_SPL_COVERAGE_CREATE_OVERALL_REPORT_IS_NECESSARY TRUE PARENT_SCOPE)
+endmacro(_spl_set_coverage_create_overall_report_is_necessary)
+
+set(COV_OUT_JSON coverage.json)
+
+function(_spl_coverage_create_overall_report)
+    if(_SPL_COVERAGE_CREATE_OVERALL_REPORT_IS_NECESSARY)
+        set(COV_OUT_VARIANT_HTML reports/coverage/index.html)
+        add_custom_command(
+            OUTPUT ${COV_OUT_VARIANT_HTML}
+            COMMAND gcovr --root ${CMAKE_SOURCE_DIR} --add-tracefile \"${CMAKE_CURRENT_BINARY_DIR}/**/${COV_OUT_JSON}\" --html --html-details --output ${COV_OUT_VARIANT_HTML}
+            DEPENDS ${GLOBAL_COMPONENTS_COVERAGE_JSON_LIST}
+            COMMENT "Generating overall code coverage report ${COV_OUT_VARIANT_HTML} ..."
+        )
+        add_custom_target(
+            unittests
+            DEPENDS coverage ${COV_OUT_VARIANT_HTML}
+        )
+        add_custom_target(
+            coverage_overall_report
+            DEPENDS ${COV_OUT_VARIANT_HTML}
+        )
+    else(_SPL_COVERAGE_CREATE_OVERALL_REPORT_IS_NECESSARY)
+        add_custom_target(unittests)
+    endif(_SPL_COVERAGE_CREATE_OVERALL_REPORT_IS_NECESSARY)
+endfunction(_spl_coverage_create_overall_report)
+
+macro(_spl_add_test_suite COMPONENT_NAME PROD_SRC TEST_SOURCES)
+    _spl_set_coverage_create_overall_report_is_necessary()
+
+    set(exe_name ${COMPONENT_NAME}_test)
+    set(PROD_PARTIAL_LINK prod_partial_${COMPONENT_NAME}.obj)
+    set(MOCK_SRC mockup_${COMPONENT_NAME}.cc)
+
+    add_executable(${exe_name}
+        ${TEST_SOURCES}
+        ${MOCK_SRC}
+    )
+
+    # Create the component library
+    add_library(${COMPONENT_NAME} OBJECT ${SOURCES})
+
+    # Define list of test specific compile options for all sources
+    # -ggdb: Produce debugging information to be able to set breakpoints.
+    # -fno-inline, -fno-default-inline, -fkeep-inline-functions:
+    # No inlining of functions to allow mocking them.
+    # -save-temps: save temporary files like preprocessed once for debugging purposes
+    set(TEST_COMPILE_OPTIONS -ggdb -fno-inline -fno-default-inline -fkeep-inline-functions -save-temps)
+
+    target_compile_options(${exe_name} PRIVATE ${TEST_COMPILE_OPTIONS})
+
+    # Coverage data is only generated for the component sources
+    target_compile_options(${COMPONENT_NAME} PRIVATE --coverage ${TEST_COMPILE_OPTIONS} ${VARIANT_ADDITIONAL_COMPILE_C_FLAGS})
```

Comment:
Please add a short how to and the reason behind using `VARIANT_ADDITIONAL_COMPILE_C_FLAGS` (e.g. add defines for compiling depending on the build kit).

Reply:
I removed this ugly interface. There is no need to define this outside of spl-core. If someone wants to add additional compile flags, she or he can do so using the common CMake commands instead of global variables.